backend-models/service-tests/sectionedservice-test.txt

```diff
@@ -0,0 +1,551 @@
+sectionedservice.php
+
+----------------------------------
+getGroupsTest() line 100:
+----------------------------------
+
+*********
+ Pre-req
+*********
+
+(checklogin) IS TRUE
+
+//// PRE-REQ Values /////
+
+send
+{
+$opt = "GRP"
+$courseid = "1"
+$coursevers = "45656"
+}
+
+/////////////////////////
+
+
+*****************
+ Compare results
+*****************
+
+If $debug is empty then successfully gathered all groups
+
+*****************
+ Service output
+*****************
+
+This service gather groups and if the groups change (unclear in code)
+this will affect:
+
+"groups",
+"grpmembershp",
+"grplst"
+
+----------------------------------
+deleteTest() line 181:
+----------------------------------
+
+*********
+ Pre-req
+*********
+
+(checklogin) IS TRUE
+AND $ha || $studentTeacher IS TRUE
+
+//// PRE-REQ Values /////
+
+send
+{
+$opt = "DEL"
+$sectid != null
+}
+
+/////////////////////////
+
+You also need to gather all listentries to compare if deleted.
+SQL Query: "SELECT FROM listentries WHERE lid=:lid" or "SELECT FROM listentries" save into beforeDeleted
+
+**************************
+ Get deleted listentries
+**************************
+
+SQL Query: "SELECT FROM listentries WHERE lid=:lid" or "SELECT FROM listentries" (depends on which was used earlier) save into afterDeleted
+
+*****************
+ Compare results
+*****************
+
+If beforeDeleted(count) > afterDeleted(count): passed
+If beforeDeleted(count) < afterDeleted(count): failed
+If $debug not empty: failed
+
+*****************
+ Service output
+*****************
+
+"codeexamples"
+
+----------------------------------
+deletedTest() line 197:
+----------------------------------
+
+*********
+ Pre-req
+*********
+
+(checklogin) IS TRUE
+AND $ha || $studentTeacher IS TRUE
+
+//// PRE-REQ Values /////
+
+send
+{
+opt = "DELETED"
+sectid != null
+}
+
+/////////////////////////
+
+You also need to gather all listentries to compare if deleted.
+SQL Query: "SELECT visible FROM listentries WHERE lid=:lid" or "SELECT visible FROM listentries" save into beforeMarkedDeleted
+
+**************************
+ Get deleted listentries
+**************************
+
+SQL Query: "SELECT FROM listentries WHERE lid=:lid" or "SELECT FROM listentries" (depends on which was used earlier) save into afterMarkedDeleted
+
+*****************
+ Compare results
+*****************
+
+If any visible has been updated to "3": passed
+If non visible has been updated to "3": failed
+If $debug not empty: failed
+
+*****************
+ Service output
+*****************
+
+Not able to identify output for listentries
+
+----------------------------------
+newCodeExampleTest() line 210:
+----------------------------------
+
+*********
+ Pre-req
+*********
+
+(checklogin) IS TRUE
+AND $ha || $studentTeacher IS TRUE
+
+//// PRE-REQ Values /////
+
+send
```

Comment:
all attributes should have a value, just stating that they are not null is not enough.
the person writing the actual test should not have to figure out what values can be used. Those values are to be documented here.